lib/rewards/server/subscribers/world-drop-subscriber.js

```diff
@@ -0,0 +1,149 @@
+    {
+        return nodes.length >= itemRewards.length;
+    }
+
+    static getTileIndex(node, worldWidth)
+    {
+        return node.x * worldWidth + node.y;
+    }
+
+    static getRewardsInteractionDistanceConfig(scene)
+    {
+        // this.range = medio tile (configurable);
+        // battle_rewards/node_modules/@reldens/skills/lib/skill.js
+        // let interactionArea = new InteractionArea();
+        // interactionArea.setupInteractionArea(this.range, targetPosition.x, targetPosition.y);
+        // let interactionResult = interactionArea.isValidInteraction(ownerPosition.x, ownerPosition.y);
+
+
+        // @TODO - BETA - rewards.interactionField
+        return scene.config.get('server/rewards/actions/interactionsDistance') ||
+            (scene.roomWorld.mapJson.tilewidth * 0.5);
+    }
+
+    static getRewardPosition(closerWalkableNodes, hasSpace)
+    {
+        if (hasSpace) {
+            //TODO: Eliminar del array de {closerWalkableNodes} la posición del enemigo (si no fue eliminada antes).
+            return closerWalkableNodes.splice(sc.randomInteger(0, closerWalkableNodes.length - 1), 1)[0];
+        } else {
+            // TODO: Caen todos en el lugar donde murió el enemigo.
+            return { x: closerWalkableNodes[0].x, y: closerWalkableNodes[0].y };
+        }
+    }
+
+    static async createRewardRoomObject(scene, targetObjectBody, reward)
+    {
+        let bodyMass = reward.hasSpace ? 1 : 0;
+        scene.objectsManager.roomObjects[reward.randomRewardId] = {
+            bodyMass: bodyMass,
+            collisionResponse: true
+            // hasState > TODO - BETA - Next feature, move drops.
+        };
+        return await scene.roomWorld.createRoomObjectBody(
+            { name: reward.randomRewardId },
+            '',
+            targetObjectBody.worldTileWidth,
+            targetObjectBody.worldTileHeight,
+            reward.rewardPosition.x,
+            reward.rewardPosition.y
+        );
+    }
+
+
+    static sendRewardsDataToClient(rewards, client)
+    {
+        let sendData = this.prepareDataToSend(rewards);
+        // @TODO - BETA - Add new event here for the sendData.
+        client.send('*', sendData);
+    }
+
+    static prepareDataToSend(rewards)
+    {
+        let hasSpace = rewards[0].hasSpace;
+        let messageData = {
+            [RewardsConst.REWARDS_CONST]: {},
+            ['hasSpace']: hasSpace
+        }
+        for (let reward of rewards) {
+            let dataReward = {
+                [RewardsConst.REWARDS_TYPE]: reward.objects_items_rewards_animations.asset_type,
+                [RewardsConst.REWARDS_KEY]: reward.objects_items_rewards_animations.asset_key,
+                [RewardsConst.REWARDS_FILE]: reward.objects_items_rewards_animations.file,
+                [RewardsConst.REWARDS_PARAMS]: reward.objects_items_rewards_animations.extra_params,
+            };
+            if (true === hasSpace) {
```

Comment:
for strict true you can just go with the variable `if(hasSpace)`, I just declare it for `false` cases because in JS `!someVariable` will take an incredible amount of things as false and we can't handle all the cases in the same way